src/components/SearchInput/index.js

```diff
@@ -15,7 +15,61 @@ class SearchInput extends Component {
 
   state = {
     value: '',
-    showAutocomplete: false
+    showAutocomplete: false,
+    listening: false,
+    voiceEnabled: false
+  };
+
+  componentDidMount() {
+    if (__CLIENT__) {
+      // eslint-disable-next-line react/no-did-mount-set-state
+      this.setState({ voiceEnabled: !!this.getSpeechRecognizer() });
+    }
+  }
+
+  shouldComponentUpdate(nextProps, newState) {
+    return [
+      this.state.voiceEnabled !== newState.voiceEnabled,
+      this.state.newState !== newState.showAutocomplete
+    ].some(test => test);
+  }
+
+  onSpeechResult = (event) => {
+    const input = this.input;
+    const text = event.results[0][0].transcript;
+
+    input.value = text;
+    this.setState({ value: text, showAutocomplete: true });
+  };
+
+  // eslint-disable-next-line no-unused-vars
+  onSpeechEnd = (event) => {
+    this.setState({ listening: false });
+    // TODO: search ?
+  };
+
+  // eslint-disable-next-line no-unused-vars
+  onSpeechError = (event) => {
+    // TODO: should we show err to user?
+  };
+
+  getSpeechRecognizer = () =>
+    window.SpeechRecognition || window.webkitSpeechRecognition;
```

Comment:
Does this work on all browsers? Can we have a check for it?

Reply:
I think there is already a library for this. Check this repo https://github.com/TalAter/annyang I can't really say how well it works though I've only used it a couple of times.